variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepositoryImpl.java

```diff
@@ -72,6 +72,12 @@ public Long countByIdsAndComplexFilters(String id, List<VariantRepositoryFilter>
         return countByComplexFiltersHelper(criteria, filters);
     }
 
+    @Override
+    public Long countTotalNumberOfVariants() {
+        return mongoTemplate.getCollection(mongoTemplate.getCollectionName(VariantMongo.class))
```

Comment:
I  think there's no need to implement this. In the `VariantWithSamplesAndAnnotationsService` just use the `count()` method that provides the CrudRepository interface (which VariantRepository implements): https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#repositories.core-concepts